Tutorial fixes

jwpeterson opened the pull request:
Misc. fixes to MOOSE training tutorial noted while teaching.

Refs #5307.

jwpeterson committed:
Tutorial: Use C++11 vector initialization in step06.

This was updated in the PackedColumn.C of step05, but not step06.

Refs #5307.

jwpeterson committed:
Fix class description of HeatConductionKernel.

jwpeterson committed:
Tutorial: Symbolically link DarcyVelocity.{C,h} files after Step 4.

These files do not change after they are introduced, so we do not need
six independent copies of them.

jwpeterson committed:
Tutorial: fix operator spacing in DarcyVelocity.C.

dschwen commented on modules/heat_conduction/src/kernels/HeatConductionTimeDerivative.C:
```diff
@@ -17,7 +17,7 @@ InputParameters validParams<HeatConductionTimeDerivative>()
   params.set<bool>("use_displaced_mesh") = true;
 
-  params.addParam<MaterialPropertyName>("specific_heat", "specific_heat",
+  params.addParam<MaterialPropertyName>("specific_heat_name", "specific_heat",
```
We actually went in the opposite direction, as it is perfectly valid to pass in a plain number in the input file and

```
specific_heat_name = 14.78
```

looks odd.

jwpeterson replied:
Well then we need to fix `density_name` in the *same* file which is the other way around.

jwpeterson replied:
And I believe there are several others within the HeatConduction module which still use the `_name` naming convention...

aeslaughter replied:
We need to pick a convention and deprecate the ones that go against it. I still like using _name because it is then clear that you are wanting a property name. In the case where you put a number, the _name indicates to me that I could use a material property. Without the _name I may think that I can't use a property and the parameter is a constant.

dschwen replied:
We do not have to shove _all_ the documentation into the parameter names themselves, we do have doc strings (and the actual parameter type is available to the UIs that help people write input files).

bwspenc replied:
I agree with @dschwen that we shouldn't have the `_name` postfix (although I recently added one myself (`eigenstrain_names`).

This isn't the only place we need some conventions on naming.  We don't have a convention on whether you put the type of the object in its name (e.g. `SideIntegralPostprocessor`).  There are lots of cases of both.  I would love to help come to some consensus on what the rules are, write them down in the Coding Standards, and then start uniformly following them.  Conventions for these kinds of things are far more important than code style conventions because this stuff breaks everyone's input files when we change our minds, while code style might be ugly, but it doesn't directly break anything.

jwpeterson commented:
> It seems that people prefer to remove "_name", then lets do that for this
> PR so at least our training is consistent.

OK, I will add a commit that does this and see if it breaks any of the apps...

jwpeterson committed:
Change diffusion_coefficient_name -> diffusion_coefficient.

Based on the discussion of #8254, the idea is that default value input
file syntax like:

specific_heat_name = 14.78

looks weird.

The goal is to drop the "_name" suffix from all the
MaterialPropertyName parameters, but this can't be done without a
deprecation period and application level changes for some names like
"density" because of their use in GlobalParams sections of input
files.

moosebuild commented:
Job `modules debug:linux-gnu` on 1b19b9e : invalidated by @dschwen

timeout in rigid body motion test
